### ISSUE TITLE
Bad Request 400 error

### ISSUE THREAD

**Issue by aytekincem** [regisb/openedx-docker#30]
After doing make all command, installation process has finished but I could not connect to the domain name i provided.  I did make lms to open the shell.  I did paver update_assets lms --settings=universal.development
./manage.py lms runserver 0.0.0.0:8000  everything seems okay however when I write http://0.0.0.0:8000/, I get bad request.

Output of terminal :

2018-06-02 19:50:11,754 INFO 970 [dd.dogapi] dog_stats_api.py:66 - Initializing dog api to use statsd: localhost, 8125
Performing system checks...

System check identified no issues (0 silenced).
June 02, 2018 - 19:50:14
Django version 1.8.18, using settings 'lms.envs.universal.production'
Starting development server at http://0.0.0.0:8000/
Quit the server with CONTROL-C.
[02/Jun/2018 19:51:21] "GET / HTTP/1.1" 400 26
[02/Jun/2018 19:51:21] "GET /favicon.ico HTTP/1.1" 400 26
[02/Jun/2018 19:51:23] "GET / HTTP/1.1" 400 26
[02/Jun/2018 19:51:23] "GET /favicon.ico HTTP/1.1" 400 26
[02/Jun/2018 19:51:35] "GET /courses HTTP/1.1" 400 26
[02/Jun/2018 19:51:35] "GET /favicon.ico HTTP/1.1" 400 26
[02/Jun/2018 19:59:46] "GET / HTTP/1.1" 400 26
[02/Jun/2018 19:59:46] "GET /favicon.ico HTTP/1.1" 400 26

**Comment by regisb** [comment]
What is the domain name you provided? And did you configure your DNS to point to your server?

**Comment by aytekincem** [comment]
Although I did the same steps when I got bad request, today I re tried doing it from shell lms and it worked. Now I am able to see openedx on my localhost. However, if I dont start it from shell and try to do it from my local repository, ( by writing 'make up' command) I see bunch of workers making some configurations however it never finishes. 
 STARTS LİKE THAT :
Attaching to openedx-docker_mongodb_1, openedx-docker_rabbitmq_1, openedx-docker_nginx_1, openedx-docker_smtp_1, openedx-docker_memcached_1, openedx-docker_elasticsearch_1, openedx-docker_mysql_1, openedx-docker_forum_1, openedx-docker_cms_1, openedx-docker_lms_1, openedx-docker_cms_worker_1, openedx-docker_lms_worker_1
mongodb_1        | 2018-06-03T15:58:27.424+0000 I CONTROL  [initandlisten] MongoDB starting : pid=1 port=27017 dbpath=/data/db 64-bit host=19ed66d11313
mongodb_1        | 2018-06-03T15:58:27.425+0000 I CONTROL  [initandlisten] db version v3.0.14
.
.
.
.
.
.
forum_1          | I, [2018-06-03T17:38:47.480201 #24]  INFO -- : worker=1 ready
forum_1          | I, [2018-06-03T17:38:47.483707 #30]  INFO -- : worker=3 ready

STUCK, NEVER FINISHES WAITED FOR HOURS BUT NO CHANGE.

I can not understand what is wrong. Any help is appreciated.

**Comment by regisb** [comment]
A couple important points

1. If you set "www.mydemo.com" as the url of your LMS with the production settings, it will only be reachable at this address. Not "localhost" and not "127.0.0.1": if you try a different url, you will get a 400 error. This is for security reasons.
2. When you run `make lms` and then `./manage.py lms runserver 0.0.0.0:8000`, the production settings are used, as indicated in the console: `using settings 'lms.envs.universal.production'`. Thus, again, the LMS is only reachable through the url you configured earlier. If you want to run a server for local testing, you have two choices:
a. Configure "localhost" as the url of the LMS during `make configure`.
b. Or run a development server with development settings: `make lms` and then `./manage.py lms --settings=universal.development runserver 0.0.0.0:8000`.
3. When you run `make up`, the services are working, but in the foreground. You can use them normally. If you really want them to run in the background (as `make all` does it), run `make daemon`.

**Comment by aytekincem** [comment]
Thank you for your help. I need to make some development on openedx. I configured the lms with the value 127.0.0.1 now when I enter the url as 127.0.0.1 it works. If I type it as localhost or localhost:8000 or 127.0.0.1:8000 it doesnt but anyway I managed to do it. My second and hopefully last question is how do I change some elements in lms. Currently I have the openedx-docker file in my Desktop, I see all the relevant files under openedx-docker/data/lms/staticfiles but of course if i simply open a .css file under staticfiles and change some style or color, the changes do not reflect on lms when I refresh 127.0.0.1. In the documentation it seems that I have to have valid development settings files in order to do this kind of changes but I couldnt understand it very well. Again thank you for your interest and it is my first project in this kind of development environment so sorry if I ask some basic stuff. Thanks.

**Comment by regisb** [comment]
Ok, I'm glad you managed to make it work. I suggest you carefully read the "Custom devstack" section from the README: https://github.com/regisb/openedx-docker#custom-devstack
You are going to clone the edx-platform repository in a separate folder, and then point to it with the EDX_PLATFORM_PATH environment variable. Then, copy the development settings from `config/openedx/universal/lms.py` to `edx-platform/lms/envs/mysettings.py`. You should be able to run standard open edx commands with these settings. For instance:

    paver update_assets lms --settings mysettings
    ./manage.py lms runserver --settings=mysettings 0.0.0.0:8000

If you are not familiar with Open edX commands, I suggest you take a look at the Open edX documentation: https://openedx.atlassian.net/wiki/spaces/OpenOPS/pages/60227787/Running+Vagrant-based+Devstack#RunningVagrant-basedDevstack-LMSWorkflow
Sometimes, Open edX can be a little... terse. When there's something you don't understand, your best bet is to just read the source code. (a pice of advice which is also relevant for running this project)

**Comment by laurayuwen** [comment]
Hi Regis, I also got 400 bad request error, when accessing using IP address.
I am using your latest release, I built the platform easily with "tutor local quickstart". I do not have a domain name, so did not provide one. I can access lms using "localhost" and cms using "studio.localhost" or "127.0.0.1".

I got 400 bad request error when I try to access it from another computer in the intranet, using the server's IP address.
